flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/sink/writer/topic/TopicProducerRegister.java

```diff
@@ -142,7 +146,8 @@ public void close() throws IOException {
             builder.topic(topic);
             Producer<T> producer = sneakyClient(builder::create);
             producerRegister.put(topic, producer);
-
+            // Set metrics for this producer
+            pulsarSinkWriterMetrics.registerSingleProducerGauges(producer);
```

Comment:
We have metrics related logic in the main code path, it can be avoided by register the gauges in the timing service, but that we need to introduce some extra fields in PulsarWriter, so I decided to make this trade-off to register metrics upon its creation.